activerecord/lib/active_record/autosave_association.rb

```diff
@@ -273,7 +282,26 @@ def changed_for_autosave?
       new_record? || has_changes_to_save? || marked_for_destruction? || nested_records_changed_for_autosave?
     end
 
+    # Returns whether or not this record is already being saved outside of the
+    # current autosave callback
+    def saving?
+      @saving
+    end
+
+    def can_save? #:nodoc:
+      !destroyed? && !saving?
+    end
+
     private
+      # Track if this record is being saved. If it is being saved we
+      # can skip saving it in the autosave callbacks
+      def saving
```

Comment:
Should this be `_` prefixed or more longer name?

IMHO `saving?`, `can_save?`, `saving` are too shorter naming for the internal methods, it can easily be overridden by the user defined methods.

Similar with 8a91423b92e6583bf1799a000616982147ca2467.

Reply:
Yes. it should. Also the documentation is missing `.` in the end of the phases. Do you want to make the changes @kamipo?

Reply:
Yeah, I'll do that.